homeassistant/components/switch/onewire.py

```diff
@@ -0,0 +1,135 @@
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the one wire Switches."""
+    base_dir = config.get(CONF_MOUNT_DIR)
+    devs = []
+    device_names = {}
+    if 'names' in config:
+        if isinstance(config['names'], dict):
+            device_names = config['names']
+
+    for family_file_path in glob(os.path.join(base_dir, '*', 'family')):
+        with open(family_file_path, "r") as family_file:
+            family = family_file.read()
+        if family in DEVICE_SWITCHES:
+            switch_id = os.path.split(
+                os.path.split(family_file_path)[0])[1]
+            for switch_key, switch_value in DEVICE_SWITCHES[family].items():
+                switch_id_complete = switch_id + "_" + switch_key
+                device_file = os.path.join(
+                    os.path.split(family_file_path)[0], switch_value)
+                devs.append(OneWireSwitch(device_names.get(switch_id_complete,
+                                                           switch_id_complete),
+                                          device_file))
+
+    if devs == []:
+        _LOGGER.error("No onewire sensor found. Check if dtoverlay=w1-gpio "
+                      "is in your /boot/config.txt. "
+                      "Check the mount_dir parameter if it's defined")
+        return
+
+    add_devices(devs, True)
+
+
+class OneWireSwitch(SwitchDevice):
+    """Representation of a OneWire switch."""
+
+    def __init__(self, name, device_file):
+        """Initialize the OneWire switch."""
+        self._name = name or DEVICE_DEFAULT_NAME
+        self._device_file = device_file
+        self._state = self._readState()
+
+    def _readState(self):
+        state = None
+        try:
+            state = self._read_value_raw() == DEVICE_SWITCH_ON
+        except ValueError:
+            _LOGGER.warning("Invalid value read from %s", self._device_file)
+        except FileNotFoundError:
+            msg = "1Wire switch file not found: %s", self._device_file
+            _LOGGER.warning(msg)
+        return state
+
+    def _read_value_raw(self):
+        """Read the value as it is returned by the switch."""
+        with open(self._device_file, 'r') as ds_device_file:
+            content = ds_device_file.read()
+        return content
+
+    @property
+    def should_poll(self):
+        """tbd, if control of OneWire Switch is not only done by hass"""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        if self._state is None:
+            self._state = self._readState()
+        return self._state
+
+    def turn_on(self, **kwargs):
+        """Turn the switch on."""
+        with open(self._device_file, 'w') as ds_device_file:
+            ds_device_file.write("1")
+        self._state = True
```

Comment:
Please introduce an update method which reads and updates the state. Don't set the state here.